grails-app/views/customer/show.gsp

```diff
@@ -85,5 +85,7 @@
                 <input type="text" name="state" value="${customer.state}" readonly>
             </div>
         </form>
+        <br>
+        <g:link controller="customer" action="update" params="[id: customer.id]"><button>Editar dados</button></g:link>
```

Comment:
```suggestion
        <g:link controller="customer" action="update" params="[id: customer.id]">Editar dados</g:link>
```
Se tu só quer criar um link pra outra página, é mais apropriado usar a tag \<a>. \<button> geralmente é usado pra disparar eventos e coisas do tipo. Se quiser deixar com aparência de botão, deixa pra quando for fazer o css.